src/probeinterface/probe.py

```diff
@@ -143,7 +144,7 @@ def model_name(self):
 
     @model_name.setter
     def model_name(self, value):
-        if value is not None:
+        if value is not None:  # Alessio, why propagate to annotations and not just set the attribute?
```

Comment:
Because annotations automatically get saved and reloaded ;)

Reply:
Plus it makes it back-compatible ;)

Reply:
I thought about the backward-compability issue after I asked. You mean that people were putting those things as annotations before right?

That makes sense.

There is one small downside though that you can't use those values as the kwargs in the sense of spike interface. That is, as a froozen sample of what arguments were passed at the beginning. 

What do I mean? When I saved those attributes the getter is called so I save let's say 'serial_number' is returned as the empty string (`''`). But when you initialize when an empty string the setter stores the new `serial_number` value as an annotation. This means that the original probe does not have an annotation for serial_number but the re-instantiaed probe does. Now that I write this I think there is an obvious solution, I will push a separate PR were we can discuss this and remove the comment from here.